### PR TITLE
Use `virtualenv` by default when restoring a model environment

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Use `virtualenv` by default when restoring a model environment.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

The default tool to restore a model environment has been changed to `virtualenv` from `conda`.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by WeichenXu123 — APPROVED** [review]

**Comment by WeichenXu123** [comment]
Q: Shall we upcoming cherry-picked master commits to branch-2.0 ? [branch-2.0](https://github.com/mlflow/mlflow/tree/branch-2.0)

**Comment by harupy** [comment]
> Q: Shall we upcoming cherry-picked master commits to branch-2.0 ? [branch-2.0](https://github.com/mlflow/mlflow/tree/branch-2.0)

I did in #6467

**Review by dbczumar — COMMENTED** [review]
Let's make sure this covers spark_udf(). Also, what about `mlflow models build-docker` with `--model-uri` specified?

File: mlflow/pyfunc/__init__.py [hunk 1]
```diff
@@ -918,7 +918,7 @@ def _get_or_create_env_root_dir(should_use_nfs):
 _MLFLOW_SERVER_OUTPUT_TAIL_LINES_TO_KEEP = 200
 
 
-def spark_udf(spark, model_uri, result_type="double", env_manager="local"):
+def spark_udf(spark, model_uri, result_type="double", env_manager=_EnvManager.VIRTUALENV):
```
Comment (harupy) [inline mlflow/pyfunc/__init__.py:921]: Do we need this change?

File: a.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { PyfuncDocUrl } from './mlflow/server/js/src/common/constants.js';
+
+console.log(PyfuncDocUrl);
```
Comment (harupy) [inline a.js:3]:
```suggestion
```

File: mlflow/pyfunc/__init__.py (hunk 1 above)
Comment (harupy) [reply mlflow/pyfunc/__init__.py]: If this is necessary, I'll create another PR.

**Review by dbczumar — APPROVED** [review]
@harupy Can we turn on `env_manager=virtualenv` by default in `mlflow.pyfunc.spark_udf()`? LGTM after that!

File: mlflow/pyfunc/__init__.py (hunk 1 above)
Comment (dbczumar) [reply mlflow/pyfunc/__init__.py]: @harupy I'd like to include this change in this PR; we should enable env restoration by default in MLflow 2.0.

Comment (harupy) [reply mlflow/pyfunc/__init__.py]: Ok, I'll update.

Comment (harupy) [reply mlflow/pyfunc/__init__.py]: updated

**Pull request merged**